### PR TITLE
add logging to file feature

### PR DESCRIPTION
**Pull request by muse-sisay**
The commit has made the following changes 
- the way csv files is read
- bumped version to `1.2`

This commit also includes added feature 
- things are logged to `keep-mega-alive.log` file
- unsuccessful logins are handled properly and logged

### CODE REVIEW

File: keep-mega-alive.sh [hunk 1]
```diff
@@ -1,21 +1,46 @@
...
+exec 4>>~/keep-mega-alive.log
+
 LOGINS=${1:-"mega-logins.csv"}
 
 if ! [ -x "$(command -v mega-version)" ]; then
 	echo 'Error: MEGAcmd is not installed. Get it from https://mega.io/cmd' >&2
 	exit 1
 fi
 
+echo -e "$(date "+%m%d%Y %T") : Starting keep-mega-alive \n" >&4
```
Comment (3ncod3) [inline keep-mega-alive.sh]: Can you change the log time format to a more universal one like [`ISO 8601`](https://www.iso.org/iso-8601-date-and-time-format.html):

```
date +"%Y-%m-%dT%H:%M:%S%:z"
```

File: keep-mega-alive.sh [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
-	echo -e "\n>>> ${LOGIN[0]}"
-	mega-login ${LOGIN[0]} ${LOGIN[1]}
-	mega-df -h
+IFS=","
+while read username password
+do
+	echo -e "\n>>> $username"
+	echo "$(date "+%m%d%Y %T") : Trying to login as $username" >&4
+	
+	mega-login "$username" "$password"
```
Comment (3ncod3) [inline keep-mega-alive.sh]: Remove the quotes around the username and password vars to make their values whitespace insensitive

File: keep-mega-alive.sh [hunk 3]
```diff
@@ -1,21 +1,46 @@
...
+	then 
+		echo "Unable to login as $username"
+		echo -e "$(date "+%m%d%Y %T") : [ERROR] Unable to login as $username \n" >&4
+		continue  
+	fi
+
+	echo "$(date "+%m%d%Y %T") : Successfully logged in as $username" >&4
+	
+	mega-df -h 
+	echo "$(date "+%m%d%Y %T") : Successfully wrote disk usage " >&4
```
Comment (3ncod3) [inline keep-mega-alive.sh]: No need to log this

File: keep-mega-alive.sh [hunk 4]
```diff
@@ -1,21 +1,46 @@
...
+	
 	mega-logout 1>/dev/null
+	echo -e "$(date "+%m%d%Y %T") : Logged out \n" >&4
```
Comment (3ncod3) [inline keep-mega-alive.sh]: Don't really need this, seems too verbose.
Also, better to have an extra new line between between runs rather logins (they're a part of the run).

File: keep-mega-alive.sh [hunk 5]
```diff
@@ -1,21 +1,46 @@
...
+do
+	echo -e "\n>>> $username"
+	echo "$(date "+%m%d%Y %T") : Trying to login as $username" >&4
+	
+	mega-login "$username" "$password"
+
+	if [ ! $? -eq  0 ]
+	then 
+		echo "Unable to login as $username"
+		echo -e "$(date "+%m%d%Y %T") : [ERROR] Unable to login as $username \n" >&4
```
Comment (3ncod3) [inline keep-mega-alive.sh]: Also log the MEGAcmd error

MEGAcmd will output something like `[API:err: 15:52:44] Failed to Login: Invalid argument` or `[API:err: 15:53:03] Login failed: invalid email or password` which would be helpful for debugging

File: keep-mega-alive.sh [hunk 6]
```diff
@@ -1,21 +1,46 @@
...
+		continue  
+	fi
+
+	echo "$(date "+%m%d%Y %T") : Successfully logged in as $username" >&4
+	
+	mega-df -h
```
Comment (3ncod3) [inline keep-mega-alive.sh]: May want to log this as well? Or at least part of it like the total used storage?

**Review by 3ncod3 — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Just a few things that need changing then we merge 👍

File: keep-mega-alive.sh (hunk 4 above)
Comment (muse-sisay) [reply keep-mega-alive.sh]: I have made the changes, new line is printed at the end to a run. But the log file seems very. "crowded".

There are time that mega-quit doesn't log out, like a connection problem. Having this printed out helps the debug process.

**Comment by muse-sisay** [comment]
My pleasure! 

Made changes as suggested, except one.

Also, I have changed the interpreter from `sh` to `bash`. I  was having problem with command substitution.

File: keep-mega-alive.sh (hunk 4 above)
Comment (3ncod3) [reply keep-mega-alive.sh]: I was using `>>>` to show mark each login like `>>> $username`

Sure we can leave that out

**Pull request merged**